### DISCUSSION TITLE
terminal emulator source directory

### DISCUSSION THREAD

**Discussion by werelord** [Murmele/Gittyup#254] [Q&A]
Trying to have Gittyup launch windows terminal as the terminal emulator command, but I'm unable to launch with the given source directory as the root..  Given Terminal's [command-line params](https://docs.microsoft.com/en-us/windows/terminal/command-line-arguments) I set: 

```
wt nt -d %1
```
terminal complains about the source folder:
```
[error 2147942667 (0x8007010b) when launching `%SystemRoot%\System32\cmd.exe']
Could not access starting directory "%1"
```

Obviously the directory isn't being passed into the command line parameter.. Is there something I'm missing?? Does the %1 (referenced in the dialog, "%1" = Repo Path) only apply to the file manager command?

**Comment by Murmele** [comment]
Hi @werelord, it is only implemented for the file manager command. You don't have to pass the working directory, because it will be done automatically: `child.setWorkingDirectory(mRepo.workdir().absolutePath());`

https://github.com/Murmele/Gittyup/blob/13377c135a56bf19f43f8527cb9fd4f7f5d55821/src/ui/RepoView.cpp
line 2686

**Reply by Epacik** [reply]
Unfortunately, it doesn't seem to work correctly if you're launching the Windows Terminal using the "wt" command using the default settings.

By default, a shell/profile opened in the Windows Terminal (or at least the ones I checked) are launched in "%USERPROFILE%" (the home directory), and not a working directory of the terminal.
That behavior can be changed by going to Settings > Profiles > [Profile you're interested in] > Starting directory.

In my opinion adding %1 to the "Terminal emulator command" option can be useful, so the people that for some reason don't want to or can't change the windows' terminal setting above could still open the terminal in repo's directory using `wt nt -d %1`

**Reply by exactly-one-kas** [reply]
Until we've implemented #299 you can use `wt nt -d .`

**Reply by Epacik** [reply]
I haven't realised that could work.
Thank you!
